SU-HSDO/suhumsci
# shs-56: add hero overlay animation for certain heroes

## Summary
This reintroduces the hero animation fixes while also accounting for the main region hero and making sure not to include the hero layered slider.

## Steps to Test
1. Make sure the Colorful theme is enabled and its option of 'Enable Animation Enhancements' is checked. 
2. Go to this page: `/components/hero-images-main-region` and edit the page.
3. Add a Hero (overlay) component to the `FULL-WIDTH REGION` at the top.
4. Add a Hero Layered Slider to the main body region.
5. Save the page and verify that the text displays for the `Hero (Gradient)` and the `Hero (Overlay)` components in the body and the main full width region. 
6. Verify the 'Hero Layered Slider' in the main body region *does not* animate. The spotlight sliders and hero no text should also not animate.
7. Edit your hero page and replace the full width region hero with the Hero Gradient slider. Verify it animates.
8. Edit your hero page again and replace the full width region hero with the 'hero layered slider'. Verify it *does not* animate.
10. Go back to the theme settings page and turn off 'Enable Animation Enhancements'. Visit your hero page and verify the slides do not animate and the text is visible. Go back and turn on animation enhancements again. 
11. Visit the hero page, turn off JS and verify the text for the heroes is visible. 
12. Change the theme to the Traditional theme and verify the heroes animate with animation enhancements on and not with animation enhancements off.
